stellargraph/core/graph.py

```diff
@@ -279,6 +274,25 @@ def __init__(
             weight_column=edge_weight_column,
         )
 
+        edge_node_set = set(self._edges.sources)
+        edge_node_set.update(self._edges.targets)
+
+        if nodes is None and edges is not None:
+            nodes = pd.DataFrame([], index=edge_node_set)
+        elif nodes is None:
+            nodes = {}
+
+        self._nodes = convert.convert_nodes(
```

Comment:
Interestingly, moving this here probably going to have to be undone if we ever do #870: we'll want `EdgeData` (`self._edges`) to be storing the source/target as ilocs, which means we need to know all of the nodes before constructing it. We shouldn't block this on that work, though.